src/cmd/script.rs

```diff
@@ -864,6 +871,13 @@ fn emit_vivado_tcl(
             .iter()
             .map(|t| (format!("TARGET_{}", t.to_uppercase()), None)),
     );
+    if let Some(define) = matches.values_of("defines") {
+        defines.extend(
+            matches.values_of("defines")
+            .unwrap()
+            .map(|t| (t.to_string(), None))
+        );
+    }
```

Comment:
Since as you say we'll need this for every script format, I would recommend outlining this code into a separate function, e.g. `fn add_defines_from_matches(defines: &mut Vec<(...)>, matches: &...)`. Then you can just call that function like `add_defines_from_matches(&mut defines, matches)` at this point. And add the call to all the other `emit_*` functions until we come up with a better scheme to streamline things.

Reply:
I implemented this, but have some questions.
1. How about the lifetime of the variables? I added the proposals the compiler gave me, but I have no clue if this makes sense.
2. Vivado seems to be the only target currently not specifying the type of the `defines` variable and the only target that uses a slightly different format... Aligning the format with other targets breaks the current implementation of parsing the source's defines. See: https://github.com/fabianschuiki/bender/pull/34/commits/32928efd6aaeeeadd065b3b4b946175112dfa790. The problem is that I would assign a reference which is only valid inside the closure. Moving this part outside the closure is probably going to break someone's flow. So, is there an easy way to convert the `defines` from `Vec<(String, Option<String>)>` to `Vec<(String, Option<&str>)>` after this closure?

Reply:
Yeah that's a bit annoying; the implementation of `separate_files_in_group` is not ideal, because it assumes ownership of `src`, which gets dropped after the call. Therefore you cannot add references from that to `defines`, which will exist after `src` was freed. Ideally `separate_files_in_group` would take `&src` instead of `src`, but let's postpone that change for the sake of getting things done.

As an alternative, I would suggest making all defines `Vec<(String, Option<String>)>`, and adding `.map(String::from)` where the compiler complains about `&str` vs `String`.

Reply:
Ok, I changed the format and it seems to be working. :tada: